identity-mdoc/src/commonMain/kotlin/com/android/identity/mdoc/transport/BleTransportCentralMdoc.kt

```diff
@@ -0,0 +1,172 @@
+package com.android.identity.mdoc.transport
+
+import com.android.identity.crypto.EcPublicKey
+import com.android.identity.mdoc.connectionmethod.ConnectionMethod
+import com.android.identity.mdoc.connectionmethod.ConnectionMethodBle
+import com.android.identity.util.Logger
+import com.android.identity.util.UUID
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.datetime.Clock
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+
+internal class BleTransportCentralMdoc(
+    override val role: Role,
+    private val options: MdocTransportOptions,
+    private val centralManager: BleCentralManager,
+    private val uuid: UUID,
+    private val psm: Int?
+) : MdocTransport() {
+    companion object {
+        private const val TAG = "BleTransportCentralMdoc"
+    }
+
+    private val mutex = Mutex()
+
+    private val _state = MutableStateFlow<State>(State.IDLE)
+    override val state: StateFlow<State> = _state.asStateFlow()
+
+    override val connectionMethod: ConnectionMethod
+        get() = ConnectionMethodBle(false, true, null, uuid)
+
+    init {
+        centralManager.setUuids(
+            stateCharacteristicUuid = UUID.fromString("00000005-a123-48ce-896b-4c76973373e6"),
+            client2ServerCharacteristicUuid = UUID.fromString("00000006-a123-48ce-896b-4c76973373e6"),
+            server2ClientCharacteristicUuid = UUID.fromString("00000007-a123-48ce-896b-4c76973373e6"),
+            identCharacteristicUuid = UUID.fromString("00000008-a123-48ce-896b-4c76973373e6"),
+            l2capUuid = if (options.bleUseL2CAP) {
+                UUID.fromString("0000000b-a123-48ce-896b-4c76973373e6")
+            } else {
+                null
+            }
+        )
+        centralManager.setCallbacks(
+            onError = { error ->
+                runBlocking {
+                    mutex.withLock {
```

Comment:
Nope, this can never happen and `BleCentralManager.setCallbacks()` already indirectly state this ("_Called if an error occurs asynchronously and the error isn't bubbled back to one of the methods on this object._"). I'll add _"Never invoked by any of the instance methods."_ to make this extra clear.